### DISCUSSION TITLE
Whisper-large latency fluctuates from 5s to 10min for the same request (only 1 minute of audio)

### DISCUSSION THREAD

**Discussion by RobZuazua** [deepgram/community#312] [General help]
### Which Deepgram product are you using?

Deepgram API

### Details

I am trying to integrate deep gram into my app workflow. 

what I have noticed in my Datadog traces is that calls to deep gram for ~1 minute of English audio can take upwards of 10 minutes. However, if I re-run the same workflow, the request completes almost immediately. 

I am wondering if there are stability issues with whisper large compared to other language models? After trying all the models, whisper large is the only one that is giving us the accuracy we need.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?punctuate=true&model=whisper-large&detect_language=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

994c54d3-1110-4461-a90d-d1b9d0cb4a88

### If possible, please attach your code or paste it into the text box.

async function submitAsyncRequest(audioSource) {
  const response = await deepgram.transcription.preRecorded(audioSource, {
    punctuate: true,
    model: "whisper-large",
    detect_language: true,
  });
  return response;
}

### If possible, please attach an example audio file to reproduce the issue.

mp3 not supported, so I can't attach it

**Comment by jjmaldonis** [comment]
Hi @RobZuazua, thanks for the detailed info. Whisper latency has been inconsistent recently and we are taking steps to improve the stability. A few weeks ago, our Whisper traffic increased dramatically. We scaled our infrastructure and applied stricter rate limiting and saw improvements immediately. Yesterday, we applied additional rate limiting to mitigate peaks in usage, and we have additional plans to improve the stability of the service tomorrow. By the middle of next week, Whisper latency should stabilize and you should consistently see "immediate" responses from our Whisper API.

**Reply by RobZuazua** [reply]
Hey @jjmaldonis, appreciate the quick response.

Please let me know once the whisper latency has stabilized.

I like the product but I cannot use this API at scale without harming reputation with my users. 

It concerns me that there have been two scaling issues in the past month that have not been communicated to users. 
- is there a status page when the API is struggling?
- are there any SLAs for the APIs?

Thanks

**Reply by jjmaldonis** [reply]
Hey @RobZuazua, we are working to solve the issues. You can find our status page here: https://status.deepgram.com/  If you have an Enterprise plan with us, that comes with an SLA.

**Reply by yousifa** [reply]
Your status page does not reflect reality. Responses are super slow > 45 sec for 1min, 600kb file and the status page shows green. When will this be fixed?

**Reply by psette** [reply]
confirmed this is still happening here. Could you guys introduce a timeout that doesn't incur any cost so we can retry?

**Reply by flogy** [reply]
I am currently evaluating switching from OpenAI to Deepgram to increase speed. Do I understand this correctly that currently Deepgram actually is sometimes much slower than OpenAI when using the whisper-large model?

Edit: I just did a comparison and unfortunately Deepgram whisper-large was about 33% slower than OpenAI and also a tiny little bit less precise. Would be great if especially the timing could be improved. On the other hand Deepgram offers a lot of functionality that OpenAI does not and has a better pricing too.
